Automattic/wp-calypso
# Switch to default theme when a paid theme is paired with the free plan.

Closes https://github.com/Automattic/wp-calypso/issues/97293

## Proposed Changes
The purpose of this PR is to include logic that will switch the users theme to the default `twentytwentyfour` when a paid theme was selected but the user chose the free plan at the plan step. This is part of the goal-first onboarding flow where users will be able to select a theme before they chose a plan. 

* Add logic to change their selected theme to the default theme when free plan is chosen with a paid theme
* Show the launchpad tasks `Select a design` as incomplete so that the user can chose to pick from the selection of free themes.

## Why are these changes being made?


*

## Testing Instructions


WIP: For a quick test we can remove `hasPaidDesign` check from the condition so it only check for the plan (This will be updated)

* Go to `/setup/onboarding` 
* Select a paid theme
* On domain page select a paid domain and continue (_this is so that we can get the modal, as it only applies to domains at the moment_)
* On the plans page select the `Start with free option` this will show the modal
* Continue with free from the modal
* After reaching the launchpad page, `twentytwentyfour` theme should be selected and the `Select a design` task incomplete.

## Pre-merge Checklist
